theano/tensor/signal/pool.py

```diff
@@ -216,7 +220,7 @@ def out_shape(imgshape, ds, ignore_border=False, st=None, padding=(0, 0)):
             elif st[0] >= ds[0]:
                 nr = (r - 1) // st[0] + 1
             else:
-                nr = max(0, (r - 1 - ds[0]) // st[0] + 1) + 1
+                nr = max(0, (r - 1 - ds[0] + st[0]) // st[0]) + 1
```

Comment:
Because in python and in C, division with negative number don't return the same value. It does the right thing for us in Python, but not in C. So in C we got wrong shapes. I prefer to use the same formula for consistency in Python and C. There is a test that make sure we keep it working.